openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
+            for(var o in tempObj) {
+                if (indicatorChildrenKey[d] == o) {
+                    var tempValue = indicatorChildrenKey[d];
+                    var tempValue2 = tempObj[o];
+                    var tempValue3 = tempObj[selectedRegion];
+                    if (primaryRiskIndicatorObj[tempValue] == undefined) {
+                        primaryRiskIndicatorObj[tempValue] = tempValue2;
+                        primaryRiskIndicatorObj.district = tempValue3;
+                    } else {
+                        primaryRiskIndicatorObj[tempValue] = primaryRiskIndicatorObj[tempValue] + "," +tempValue2;
+                        primaryRiskIndicatorObj.district = primaryRiskIndicatorObj.district + "," +tempValue3;
+                    }
                 }
             }
-            if (grids != undefined) {
-                layerNames[grids] = [];
-            }
         }
-
-        // Create the category list (population the object)
-        for (var j=0; j < json.length; j++) {
-            name = json[j].mapped_value;
-            cat = json[j].category;
-            type = json[j].type;
-            grids = json[j].grids;
-            pDef = json[j].project_definition;
-
-            if (cat != undefined && type == 'svir-qgis') {
-                layerId = json[j].id;
-                layerTitle = json[j].mapped_value;
-                layerNames[name].push(layerId);
-                layersByCat[cat].push(layerTitle);
-                projectDefinition[name].push(pDef);
+    }
+    return primaryRiskIndicatorObj;
+}
+
+function createIndex(la, index) {
+    var ct = 0;
+    var indicator = [];
+    // setup the indicator with all the municipalities
+    for (var ia = 0; ia < la.length; ia++) {
+        var temp = {};
+        temp.region = la[ia].properties[selectedRegion];
+        indicator.push(temp);
+    }
+    for (var i = 0; i < index.length; i++) {
+        for (var j = 0; j < la.length; j++, ct++) {
+            if (indicator[j].region == la[j].properties[selectedRegion]) {
+                var tempName = index[i].name;
+                var tempValue = la[j].properties[tempName];
+                indicator[j][tempName] = tempValue;
             }
         }
-
-        // Get unique category names
-        var categoryUnique = categoryList.filter(function(itm,i,categoryList){
-            return i==categoryList.indexOf(itm);
-        });
-
-        for (var l in categoryUnique) {
-            // Append category names to dropdown list
-            var categoryTitle = categoryUnique[l];
-            var opt = document.createElement('option');
-            opt.innerHTML = categoryTitle;
-            opt.value = categoryTitle;
-            selCat.appendChild(opt);
-            // Append layer list to dowpdown
-            var layerOpt = document.createElement('option');
+    }
+    return indicator;
+}
+
+function combineIndicators(nameLookUp, themeObj, JSONthemes) {
+    projectDef = sessionProjectDef;
```

Comment:
yes, sessionProjectDef is the project definition as is was when uploaded from the QGIS tool, bit projectDef includes modified weights and is no longer the version that was uploaded from the QGIS tool.